packages/tests/tests/sdk/content_topic.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { wakuFilter } from "@waku/core";
+import {
+  bytesToUtf8,
+  createEncoder,
+  createLightNode,
+  DEFAULT_CLUSTER_ID,
+  defaultLibp2p,
+  LightNode,
+  Protocols,
+  streamContentTopic,
+  subscribeToContentTopic,
+  utf8ToBytes,
+  waitForRemotePeer,
+  WakuNode
+} from "@waku/sdk";
+import {
+  contentTopicToPubsubTopic,
+  ensureShardingConfigured,
+  pubsubTopicToSingleShardInfo
+} from "@waku/utils";
+import { expect } from "chai";
+
+import { makeLogFileName, ServiceNode, tearDownNodes } from "../../src";
+
+describe("SDK: Creating by Content Topic", function () {
+  const ContentTopic = "/myapp/1/latest/proto";
+  const testMessage = "Test123";
+  let nwaku: ServiceNode;
+  let waku: LightNode;
+  let waku2: LightNode;
+
+  beforeEach(async function () {
+    this.timeout(15000);
+    nwaku = new ServiceNode(makeLogFileName(this) + "1");
+    await nwaku.start({
+      pubsubTopic: [contentTopicToPubsubTopic(ContentTopic)],
+      lightpush: true,
+      relay: true,
+      filter: true,
+      discv5Discovery: true,
+      peerExchange: true,
+      clusterId: DEFAULT_CLUSTER_ID
+    });
+  });
+
+  afterEach(async function () {
+    this.timeout(15000);
+    await tearDownNodes(nwaku, [waku, waku2]);
+  });
+
+  it("given a content topic, creates a waku node and filter subscription", async function () {
+    const expectedPubsubTopic = contentTopicToPubsubTopic(ContentTopic);
+
+    waku = (
+      await subscribeToContentTopic(ContentTopic, () => {}, {
+        peer: await nwaku.getMultiaddrWithId()
+      })
+    ).waku;
+
+    expect((waku as WakuNode).pubsubTopics).to.include(expectedPubsubTopic);
```

Comment:
why are we having to typecast it `as WakuNode`? are there any problems with implicit inference?